### PR TITLE
Add sync functionality to re-queue until the Status is Active

### PR DESCRIPTION
**Pull request by Vandita2020**
Re-queue when function `State` is not `Active`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/e2e/tests/test_function.py [hunk 1]
```diff
@@ -420,3 +420,59 @@ def test_function_package_type_image_with_signing_config(self, lambda_client):
 
         # Check Lambda function doesn't exist
         assert not lambda_validator.function_exists(resource_name)
+
+    def test_function_reconciler(self, lambda_client):
```
Comment (A-Hilaly) [inline test/e2e/tests/test_function.py:424]: Can you name this `test_function_is_synced` ?

Comment (Vandita2020) [reply test/e2e/tests/test_function.py:424]: Sure

Comment (Vandita2020) [reply test/e2e/tests/test_function.py]: Done

**Review by A-Hilaly — APPROVED** [review]
Thank you @Vandita2020 
/lgtm

**Pull request merged**